Remediation fails when Sonarr/Radarr recycling bin is enabled

Hey, I received this error after my first corrupt file failed:

```
2026-06-19T22:08:28-04:00 [INFO] Corruption detected in file: /plexmedia/TV Shows/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv (Type: CorruptHeader)
2026-06-19T22:08:28-04:00 [INFO] Skipping duplicate corruption for file already being processed: /plexmedia/TV Shows/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv
2026-06-19T22:09:24-04:00 [INFO] Handling corruption for file: /plexmedia/TV Shows/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv
2026-06-19T22:09:24-04:00 [INFO] Auto-remediation enabled for /plexmedia/TV Shows/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv, proceeding immediately
2026-06-19T22:09:24-04:00 [INFO] Parse failed, falling back to listing all media for sonarr
2026-06-19T22:09:24-04:00 [INFO] Matched media: Jury Duty (ID: 693)
2026-06-19T22:09:24-04:00 [INFO] Deleting file ID 174475 from sonarr
2026-06-19T22:09:54-04:00 [INFO] *arr API request failed (attempt 1/3): Delete "http://192.168.1.11:8989/sonarr/api/v3/episodefile/174475": context deadline exceeded (Client.Timeout exceeded while awaiting headers), retrying...
2026-06-19T22:10:26-04:00 [INFO] *arr API request failed (attempt 2/3): Delete "http://192.168.1.11:8989/sonarr/api/v3/episodefile/174475": context deadline exceeded (Client.Timeout exceeded while awaiting headers), retrying...
2026-06-19T22:10:30-04:00 [ERROR] Failed to delete file /tv/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv: failed to delete file: 404 Not Found
2026-06-19T22:11:43-04:00 [INFO] Handling corruption for file: /plexmedia/TV Shows/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv
2026-06-19T22:11:43-04:00 [INFO] Auto-remediation enabled for /plexmedia/TV Shows/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv, proceeding immediately
2026-06-19T22:11:43-04:00 [INFO] Parse failed, falling back to listing all media for sonarr
2026-06-19T22:11:43-04:00 [INFO] Matched media: Jury Duty (ID: 693)
2026-06-19T22:11:43-04:00 [ERROR] Failed to delete file /tv/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv: file not tracked in sonarr; cannot remediate via API: /tv/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv
```

After troubleshooting, it seems Sonarr takes a minute to move the file to the recycle bin. In that time, healarr times out. Retrying just leads to:

`2026-06-19T22:34:42-04:00 [ERROR] Failed to delete file /tv/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv: file not tracked in sonarr; cannot remediate via API: /tv/Jury Duty {tmdb-222023}/Season 2/Jury Duty.S02E07.Mergers and Acquisitions [WEBDL-1080p]-NTb.mkv`

Since by then the file has been deleted from Sonarr, but healarr doesn't trigger a new search. Disabling the recycling bin is a band-aid fix. Maybe an option to increase the wait time would fix this?

Also, does Healarr have the ability to blocklist a bad download? Otherwise the same corrupt file just gets downloaded again.

--- a/internal/integration/arr_client.go
+++ b/internal/integration/arr_client.go
@@ -793,6 +793,22 @@ func (c *HTTPArrClient) deleteFileByID(instance *ArrInstance, fileID int64) erro
 	}
 	defer resp.Body.Close()
 
+	// 404 means the file ID we asked to delete no longer exists on the *arr,
+	// which is exactly the deletion goal - treat it as success, not failure.
+	// This is the recycle-bin race (issue #350): with a *arr recycling bin
+	// enabled, the DELETE can take ~1 minute while the *arr moves the file,
+	// so the request times out client-side; the *arr finishes the move
+	// server-side, and our retry then hits 404. Reporting that as a delete
+	// failure stalled remediation - the file WAS gone, but no replacement
+	// search was ever triggered. We only reach this function after resolving
+	// a concrete fileID from the *arr's own file list, so a 404 here is
+	// unambiguous (unlike a file that was never tracked, which is handled
+	// separately and deliberately refused).
+	if resp.StatusCode == http.StatusNotFound {
+		logger.Infof("File ID %d already absent from %s (404 on delete) - treating as deleted", fileID, instance.Type)
+		return nil
+	}
+
 	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
 		return fmt.Errorf("failed to delete file: %s", resp.Status)
 	}

--- a/internal/integration/arr_client_test.go
+++ b/internal/integration/arr_client_test.go
@@ -1029,6 +1029,55 @@ func TestHTTPArrClient_DeleteFile_Radarr(t *testing.T) {
 	}
 }
 
+// Recycle-bin race (issue #350): the file is tracked in the *arr (so a fileID
+// resolves), but the DELETE returns 404 because the *arr already moved it to
+// its recycling bin while our request timed out and retried. That 404 means
+// the file is gone - the deletion goal - so DeleteFile must succeed and return
+// metadata so the remediator proceeds to search for a replacement, NOT fail.
+func TestHTTPArrClient_DeleteFile_404OnDeleteIsSuccess(t *testing.T) {
+	client, db := setupTestClient(t)
+	defer db.Close()
+
+	deleteEndpointCalled := false
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch {
+		case r.URL.Path == "/api/v3/episodefile" && r.Method == "GET":
+			// File IS still tracked when we list (the recycle-bin move
+			// hasn't been reflected in the file list yet).
+			json.NewEncoder(w).Encode([]struct {
+				ID   int64  `json:"id"`
+				Path string `json:"path"`
+			}{
+				{ID: 174475, Path: "/tv/Show/Season 02/ep07.mkv"},
+			})
+		case r.URL.Path == "/api/v3/episodefile/174475" && r.Method == "DELETE":
+			// ...but by the time the delete lands, the *arr already removed it.
+			deleteEndpointCalled = true
+			w.WriteHeader(http.StatusNotFound)
+		case strings.HasPrefix(r.URL.Path, "/api/v3/episode"):
+			json.NewEncoder(w).Encode([]struct{}{})
+		default:
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+	defer server.Close()
+
+	encryptedKey, _ := crypto.Encrypt("key")
+	db.DB.Exec(`INSERT INTO arr_instances (id, name, type, url, api_key, enabled) VALUES (1, 'Sonarr', 'sonarr', ?, ?, 1)`, server.URL, encryptedKey)
+	db.DB.Exec(`INSERT INTO scan_paths (id, local_path, arr_path, arr_instance_id, auto_remediate, is_4k) VALUES (1, '/local/tv', '/tv', 1, 0, 0)`)
+
+	metadata, err := client.DeleteFile(693, "/tv/Show/Season 02/ep07.mkv")
+	if err != nil {
+		t.Fatalf("DeleteFile must treat a 404-on-delete as success (file is gone): %v", err)
+	}
+	if !deleteEndpointCalled {
+		t.Error("delete endpoint was not called")
+	}
+	if metadata == nil {
+		t.Error("metadata must be returned so the remediator can trigger a search")
+	}
+}
+
 func TestHTTPArrClient_DeleteFile_NotFoundInArr(t *testing.T) {
 	client, db := setupTestClient(t)
 	defer db.Close()

